### ISSUE TITLE
Changes the past on the Number of outstanding reports graph after storage

### ISSUE THREAD

**Issue by csordasmarton** [Ericsson/codechecker#3622]
CodeChecker changes the past on the following scenario (introduce reports with same report hashes):

- Start a web server with the master branch.
- Analyze and store the following source file (it should find 2 reports -> `bugprone-sizeof-expression` and `clang-diagnostic-unused-value`):
```cpp
int main() {
  sizeof(42);
}
```
- Open the database and change the detection date manually 2 days earlier (this way it will be easier to see the problem).
- Suppress this report with a source code comment, reanalyze the code and store it again to the same run:
```cpp
int main() {
  // codechecker_suppress [bugprone-sizeof-expression] Test
  sizeof(42);
}
```
- Check the number of outstanding reports chart and it shows correct values:
![image](https://user-images.githubusercontent.com/6695818/152799075-cb486e75-fcee-4f04-8bcc-60ee8798c95a.png)

- Change the surce code and introduce a new finding with the same hash:
```cpp
int main() {
  // codechecker_suppress [bugprone-sizeof-expression] Test
  sizeof(42);

  sizeof(42);
}
```
- Analyze and store the results to the same run and check the chart again:
![image](https://user-images.githubusercontent.com/6695818/152799410-19b1237d-d270-462a-8a8b-385ff584d762.png)

:exclamation: With this I **changed the past**. On the first screenshot we can see that on 5th of February I had 1 HIGH report but now I have two.

**Comment by bruntib** [comment]
This issue has been solved by now.
